osidb/mixins.py

```diff
@@ -555,6 +559,46 @@ def unembargo(self):
             # continue deeper into the related context
             related_instance.unembargo()
 
+    def set_public_nested(self):
+        """
+        Change internal ACLs to public ACLs for all related Flaw objects and save them.
+        The only exception is "snippets", which should always have internal ACLs.
+        The Flaw itself will be saved later to avoid duplicate operations.
+        """
+        from osidb.models import Flaw
+
+        if not isinstance(self, Flaw):
+            if not self.is_internal:
+                return
+            kwargs = {}
+            if issubclass(type(self), AlertMixin):
+                # suppress the validation errors as we expect that during
+                # the update the parent and child ACLs will not equal
+                kwargs["raise_validation_error"] = False
+            if issubclass(type(self), TrackingMixin):
+                # do not auto-update the updated_dt timestamp as the
+                # followup update would fail on a mid-air collision
+                kwargs["auto_timestamps"] = False
+            self.set_public()
+            self.save(**kwargs)
+
+        # chain all the related instances as we
+        # only care for the ACLs which are unified
+        for related_instance in chain.from_iterable(
+            getattr(self, name).all()
+            for name in [
+                related.related_name
+                for related in self._meta.related_objects
+                # only the models with ACLs other than "snippets" are subject of this
+                if (
+                    issubclass(related.related_model, ACLMixin)
+                    and related.related_name != "snippets"
+                )
+            ]
+        ):
+            # continue deeper into the related context
+            related_instance.set_public_nested()
```

Comment:
Correct me if I am wrong but this seems to potentially change the tracker visibility too. There are however PS modules (and it is quite frequent - eg. https://gitlab.cee.redhat.com/prodsec/product-definitions/-/blob/master/data/platforms/ps_modules.json?ref_type=heads#L92) where even when the flaw is public the trackers should stay internal. Just blindly setting all the related stuff public would break this assumption.

Reply:
So after discussing this I found out that my comment was actually not relevant. As there is no internal visibility level defined for the trackers in Bugzilla/Jira we just use the public one from the product definitions. Then, the change from internal to public in OSIDB is actually a no-op for the tracker visibility in there.